### ISSUE TITLE
[Map] LOIs not loaded from local db

### ISSUE THREAD

**Issue by gino-m** [google/ground-android#1358]
The LOI in question has a valid geometry in the remote db, so it appears geometries are not getting written and/or read to the local db.

```
2022-12-22 18:25:03.697 6294-6554/com.google.android.ground E/StreamUtil: com.google.android.ground.persistence.local.LocalDataConsistencyException: No geometry in location of interest LocationOfInterestEntity(id=zg2aaoSkxCdhgTGuqTil, surveyId=fhJ4uGV2Vh2gkcaVh48v, jobId=KSbv3D9O9xx0kNkfOTGz, state=DEFAULT, created=AuditInfoEntity(user=UserDetails{id=, email=, displayName=Unknown user}, clientTimestamp=0, serverTimestamp=0), lastModified=AuditInfoEntity(user=UserDetails{id=, email=, displayName=Unknown user}, clientTimestamp=0, serverTimestamp=0), geometry=null).id
        at com.google.android.ground.persistence.local.room.converter.ConverterExtKt.toModelObject(ConverterExt.kt:201)
        at com.google.android.ground.persistence.local.room.RoomLocalDataStore.toLocationsOfInterest$lambda-13$lambda-12(RoomLocalDataStore.kt:214)
        at com.google.android.ground.persistence.local.room.RoomLocalDataStore.$r8$lambda$m3ISXtF1zWwNXRYocAUBSwtnl6k(Unknown Source:0)
        at com.google.android.ground.persistence.local.room.RoomLocalDataStore$$ExternalSyntheticLambda53.get(Unknown Source:4)
        at com.google.android.ground.util.StreamUtil.logErrorsAndSkipKt(StreamUtil.kt:42)
        at com.google.android.ground.persistence.local.room.RoomLocalDataStore.toLocationsOfInterest(RoomLocalDataStore.kt:214)
        at com.google.android.ground.persistence.local.room.RoomLocalDataStore.getLocationsOfInterestOnceAndStream$lambda-11(RoomLocalDataStore.kt:206)
        at com.google.android.ground.persistence.local.room.RoomLocalDataStore.$r8$lambda$dYl4m1PNuXkXj_V641VN7MvGZk8(Unknown Source:0)
        at com.google.android.ground.persistence.local.room.RoomLocalDataStore$$ExternalSyntheticLambda20.apply(Unknown Source:6)
        at io.reactivex.internal.operators.flowable.FlowableMap$MapSubscriber.onNext(FlowableMap.java:63)
        at com.akaita.java.rxjava2debug.extensions.FlowableOnAssembly$OnAssemblySubscriber.onNext(FlowableOnAssembly.java:61)
        at io.reactivex.internal.operators.flowable.FlowableFlatMapMaybe$FlatMapMaybeSubscriber.innerSuccess(FlowableFlatMapMaybe.java:175)
        at io.reactivex.internal.operators.flowable.FlowableFlatMapMaybe$FlatMapMaybeSubscriber$InnerObserver.onSuccess(FlowableFlatMapMaybe.java:397)
        at com.akaita.java.rxjava2debug.extensions.MaybeOnAssembly$OnAssemblyMaybeObserver.onSuccess(MaybeOnAssembly.java:68)
        at io.reactivex.internal.operators.maybe.MaybeFromCallable.subscribeActual(MaybeFromCallable.java:61)
        at io.reactivex.Maybe.subscribe(Maybe.java:4290)
        at com.akaita.java.rxjava2debug.extensions.MaybeOnAssemblyCallable.subscribeActual(MaybeOnAssemblyCallable.java:43)
        at io.reactivex.Maybe.subscribe(Maybe.java:4290)
        at io.reactivex.internal.operators.flowable.FlowableFlatMapMaybe$FlatMapMaybeSubscriber.onNext(FlowableFlatMapMaybe.java:132)
        at com.akaita.java.rxjava2debug.extensions.FlowableOnAssembly$OnAssemblySubscriber.onNext(FlowableOnAssembly.java:61)
        at io.reactivex.internal.operators.flowable.FlowableObserveOn$ObserveOnSubscriber.runAsync(FlowableObserveOn.java:407)
        at io.reactivex.internal.operators.flowable.FlowableObserveOn$BaseObserveOnSubscriber.run(FlowableObserveOn.java:176)
        at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker$BooleanRunnable.run(ExecutorScheduler.java:288)
        at io.reactivex.internal.schedulers.ExecutorScheduler$ExecutorWorker.run(ExecutorScheduler.java:253)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
        at java.lang.Thread.run(Thread.java:764)
```

**Comment by gino-m** [comment]
This appears to be because POINT and POLYGON are supported by `ConverterExt`; we need to also add support for `MULTI_POLYGON`. Also, rather than throwing an exception, it might be safer to catch the error, skip the LOI, and log a warning.

@scolsen Could you PTAL?

**Comment by JSunde** [comment]
How do we want to convert MultiPolygons to GeometryEntities? Should a GeometryEntity have a nullable List<GeometryEntity> so that the top level GeometryEntity represents the MultiPolygon and the sub GeometryEntities would be the individual polygons that make it up?

Or maybe the vertices and holes fields should be List<String?> where each string represents the vertices and holes of the child polygons?

WDYT @gino-m @scolsen?

**Comment by scolsen** [comment]
I think everything is in place except for the local data store conversion function. we need to add the conversions here:

```
fun GeometryEntity.toModelObject() =
  when (geometryType) {
    GeometryType.POINT.name -> this.toPointModel()
    GeometryType.POLYGON.name -> this.toPolygonModel()
    else -> null
  }

fun Geometry.toLocalDataStoreObject() =
  when (this) {
    is Point -> this.toLocalDataStoreObject()
    is Polygon -> this.toLocalDataStoreObject()
    else -> null
  }
```

the data format is already defined by the converter:

```
private fun getMultiPolygonCoordinates(
    multiPolygon: MultiPolygon
  ): IndexedMap<IndexedMap<IndexedMap<GeoPoint>>> =
    listToIndexedMap(multiPolygon.polygons.map(this::getPolygonCoordinates))
```

So we need to implement the to/from converter that takes the model object representation and spits out that gson representation. You should be able to implement it in terms of the parseVertices and other methods already in the ConverterExt.kt -- but it might need some additional helpers. THe model object is just a list of polygon so in theory it's just a matter of calling `toLocalDtaStore` on each polygon then joining the resulting list into a single Gson map

**Comment by JSunde** [comment]
I think this should be fixed after #1431
